actions/deprovision_mailing_lists.py

```diff
@@ -0,0 +1,117 @@
+"""
+See docstring of deprovision_mailing_lists()
+
+Remove from KeyCloak mailing list groups (the subgroups of the given mailing
+list group tree root) users who are not members those groups' respective
+`allow_members_from_group_trees` attributes. Users that have the attribute
+`institutionless_since` set will be given the provided grace period before
+removal.
+
+The grace period is intended to avoid removal of users who are changing
+institutions (i.e. when a user is removed from institution A before being
+added to institution B). Note that users who are part of multiple institutions
+will be removed immediately, since they will not have `institutionless_since`
+set after being removed from only one institution.
+
+Similarly, it is important that whatever updates `institutionless_since` is
+run before this action. Otherwise when a user is removed from their last
+institution, they will no longer satisfy the `allow_members_from_group_trees`
+constraint, but their `institutionless_since` won't be set. (This is a race
+condition, so the grace period mechanism isn't 100% reliable.)
+
+Example::
+
+       python -m actions.deprovision_mailing_lists --ml-group-root /test-mail
+"""
+import asyncio
+import logging
+
+from datetime import datetime, timedelta
+
+from krs.token import get_rest_client
+from krs.groups import get_group_membership, group_info, remove_user_group
+from krs.users import user_info
+
+
+logger = logging.getLogger('deprovision_mailing_lists')
+
+
+async def get_group_tree_membership(root_group_path, ignore_subgroup_names, keycloak_client):
```

Comment:
This function should probably be cached, since I assume it will be called for every mailinglist with basically the same args.  Something like
https://pypi.org/project/asyncache/